server/backbone/notifications.py

```diff
@@ -0,0 +1,52 @@
+from pyfcm import FCMNotification
+from abc import ABC, abstractmethod
+from .models import Parent, Child
+
+
+class Notification(ABC):
+
+  api_key = "AAAAuKgxco4:APA91bF0GiF-EGkdRLYXzZYIf76YQffw_93_GTut2N2pj48CS_Vdw44scWBzJqkobei5xcfE9IET2mXLIOGQJdOGQbhkal_A9BYfQNQ3L1AWcZnsIQdfoiku2htuETMOB2qzeehBQ2hE"
+  push_service = FCMNotification(api_key=api_key)
+ 
+  def __init__(self, receiver, message_title, message_body):
+    self.receiver = receiver
+    self.message_title = message_title
+    self.message_body = message_body
+    super().__init__()
+
+  def should_send_notification(self):
+    return self.receiver.registration_id and self.receiver_should_get_notification()
+  
+  def send_notification(self):
+    if (self.should_send_notification()):
+      return self.push_service.notify_single_device(registration_id=self.receiver.registration_id, message_title=self.message_title, message_body=self.message_body)
+    else:
+      return False
+
+  @abstractmethod
+  def receiver_should_get_notification(self):
+    pass
+
+
+class NewQuestNotification(Notification):
+
+  def __init__(self, receiver, quest_name):
+    message_title = "New Quest"
+    message_body = "You have a new quest to complete: " + quest_name
+    self.receiver = receiver
+    super().__init__(receiver, message_title, message_body)
+
+  def receiver_should_get_notification(self):
+    return isinstance(self.receiver, Child)
```

Comment:
It's good practice to keep logic for who should receive notifs within the notif's class so we don't accidentally send any to users who shouldn't receive them.
Sure the class isn't 100% necessary but if notifications get any more complex or if Jim wants to add any in the future when working on the app it will be a lot easier

Reply:
It's good, but:
1. Your check is wrong, I notice that I have not updated the decorators. I pushed my fix for that now onto this PR.
2. What's the point of this check if you have guaranteed child account on the end-point by the `child_login_required` decorator?

Reply:
wachu mean the check is wrong?

and it's good to check in case other things ever break, that's how notifications should be

Reply:
This check is redundant. Things will never be that broken.
If you wanna have a check here then see my last commit on this branch to see how to actually check for user type.

Reply:
Ok, but also if we were to expand and use this notif somewhere else there would not necessarily be a check so it's always good to have. This method is also where we could check the user's notification settings so I'd rather leave things as they are. 
But yes I can fix the check